### ISSUE TITLE
Add validation for mappings

### ISSUE THREAD

**Issue by nickmoorman** [cloudtools/troposphere#426]
First off, this library has been a godsend.  Nice work!  =D

I was working on a template today, and I added a mapping like this (pardon the grossness...):

    template.add_mapping("HyperkubeImage", {
        "official": {
            "v1.1.2": "gcr.io/google_containers/hyperkube:v1.1.2",
            "v1.1.4": "gcr.io/google_containers/hyperkube:v1.1.4",
            "v1.1.7": "gcr.io/google_containers/hyperkube:v1.1.7",
            "v1.1.8": "gcr.io/google_containers/hyperkube:v1.1.8",
            "v1.2.0-beta.0": "gcr.io/google_containers/hyperkube:v1.2.0-beta.0"
        },
        "coreos": {
            "v1.1.7-coreos.1": "quay.io/coreos/hyperkube:v1.1.7-coreos.1",
            "v1.1.8-coreos.0": "quay.io/coreos/hyperkube:v1.1.8-coreos.0"
        }
    })

The final JSON was generated just fine, but when I tried to pass it to CloudFormation from the AWS CLI, I got the error:

    A client error (ValidationError) occurred when calling the CreateStack operation:
      Template format error: Mappings attribute name 'v1.1.2' must contain only alphanumeric characters.

It would be nice if Troposphere could catch these sort of errors prior to rendering the template.

**Comment by markpeek** [comment]
Agreed we should add attribute name validation to mappings. Although I see the statement "must contain only alphanumeric characters" but AWS [docs](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/mappings-section-structure.html) show the use of a "-" character as well.

**Comment by nickmoorman** [comment]
I think the alphanumeric restriction only holds true for the inner mapping elements; restructuring my mapping like this works fine (although it needs to be reworked completely at this point):

    template.add_mapping("HyperkubeImage", {
        "v1.1.2": {
            "official": "gcr.io/google_containers/hyperkube:v1.1.2"
        },
        "v1.1.4": {
            "official": "gcr.io/google_containers/hyperkube:v1.1.4"
        },
        "v1.1.7": {
            "official": "gcr.io/google_containers/hyperkube:v1.1.7"
        },
        "v1.1.8": {
            "official": "gcr.io/google_containers/hyperkube:v1.1.8"
        },
        "v1.2.0-beta.0": {
            "official": "gcr.io/google_containers/hyperkube:v1.2.0-beta.0"
        },
        "v1.1.7-coreos.1": {
            "coreos": "quay.io/coreos/hyperkube:v1.1.7-coreos.1"
        },
        "v1.1.8-coreos.0": {
            "coreos": "quay.io/coreos/hyperkube:v1.1.8-coreos.0"
        }
    })

I've also seen plenty of mappings where the outer keys are AWS regions (`us-east-1`, etc.).

**Comment by nickmoorman** [comment]
While we're on the subject, I just discovered another validation that could be added.  The top-level mapping keys (such as `v1.1.8-coreos.0` in my latest comment) apparently have undocumented requirements:

    A client error (ValidationError) occurred when calling the CreateStack operation:
        Template format error: Mappings element name 'v1.1.8_coreos.0' must be non-empty and can contain only alphanumerics, '-' or '.'

I believe that would make the final validation requirements for a mapping something like:

    {
        "([A-Za-z0-9]|-|\.)+": {
            "([A-Za-z0-9])*": "No known requirements?"
        }
    }
